### DISCUSSION TITLE
Directing JC output into new Python variable?

### DISCUSSION THREAD

**Discussion by patrik-schoon-axelsson** [kellyjonbrazil/jc#165] [Q&A] [answered]
Hi there,

I'm trying to integrate JC into a Flask-server that'll provide a REST endpoint from where I can get a JSON return of IPTables. I'm still finding my feet in Linux/BASH, so I thought I'd ask you directly in the repo here. The trouble is this line:

iptables = os.system("sudo iptables -L | jc -r --iptables")

This variable prints the output of my IPtables to the terminal, which is great. However, to make it work with my plans for it, I need it to return the JSONified output as JSON, insted of returning the exit-code 0 it produces right now. Is there some method for extracting the command output into a separate variable or a flag to return something other than the exit code on execution that I'm missing? Or should I try to pipe the output into some further tool, if I just want to work with the output JSON and not the exit code?

Your help is appreciated and my thanks for a great tool!

**Comment by kellyjonbrazil** [comment, marked as answer]
Hi there - thanks for your question!

There are a few ways to tackle this, but I think this is probably the cleanest approach.

Instead of using the `os.system` library, you can use the `subprocess` library to run a command and capture the output. You also do not need to pipe the output through `jc` within the `subprocess` call because `jc` is also a Python library and can be used directly in the program. Here's how you can grab the output and parse it:
```
import subprocess
from jc.parsers.iptables import parse as parse_iptables

cmd_output = subprocess.check_output(['sudo', 'iptables', '-L'], text=True)
iptables_data = parse_iptables(cmd_output, quiet=True, raw=True)
```
Because you are using the `jc` parser library directly, there is no need to deserialize the JSON output to a dictionary. The parser will return a dictionary (or list of dictionaries, depending on the parser) directly.

Hope that helps!

**Reply by kellyjonbrazil** [reply]
Here's the new-style way to use `jc` in Python scripts since v1.18.0:
```
import subprocess
import jc

cmd_output = subprocess.check_output(['sudo', 'iptables', '-L'], text=True)
iptables_data = jc.parse('iptables', cmd_output, quiet=True, raw=True)
```
The old way still works, but the new higher-level API allows you to access builtin and plugin parsers the same way.
